scripts/TRN(title matching).R

```diff
@@ -0,0 +1,139 @@
+
+# If there is no official title available in ClinicalTrials.gov, the brief title will be used as the 'title'
+IV_updated_titles <- left_join(IV_ids, NCT_full_titles, by = "id") |>
+  mutate(title = ifelse(is.na(official), brief, official)) |>
+  select(-brief, -official)
+
+# Process IV titles to make compatible with title matching algorithm
+IV_updated_titles <- IV_updated_titles |>
+  mutate(title_processed = tolower(title) |>
+           stringr::str_squish() |> # remove whitespace at start and end, as well as any "\t" whitespace characters
+           stringr::str_remove_all("[:punct:]") |>
+           stringr::str_remove_all(" ")) |>
+  select(id, title, title_processed)
+
+###########################################################
+
+# Extract EU IDs and country data (for filtering), plus process EU titles to be used in title matching algorithm
+EU_ids <- EU_dump |>
+          select(eudract_number,
+                 member_state_concerned,
+                 full_title_of_the_trial,
+                 sponsors)
+
+EU_ids <- rename(EU_ids, id = eudract_number, state = member_state_concerned, title = full_title_of_the_trial)
+
+# Filter out EU trials which dont have 'Germany' listed in member_state_concerned
+# Yields table with 13068 unique trials which mention Germany in this field
+# Can filter however we want here.
+
+EU_only_German <- EU_ids |>
+                  filter(grepl('Germany', state))
+
+# Process EU titles
+EU_only_German <- EU_only_German |> select(id,title) |>
+  tidyr::drop_na(title) |>
+  distinct(id, title,.keep_all = TRUE) |>
+  mutate(title_processed = tolower(title) |>
+           stringr::str_squish() |> # remove whitespace at start and end, as well as any "\t" whitespace characters
+           stringr::str_remove_all("[:punct:]") |>
+           stringr::str_remove_all(" "))
+
+##########################################################
+## Title matching (from 02_cross-reg-title-matching.R)
+
+# assigning maxDist value for title matching (Elements in x will not be matched
+# with elements of table if their distance is larger than maxDist)
+DISTANCE <- 7
+
+## The below takes about 3.5 hours on my baseline 2020 MacBook Air
+## Let it run, it takes a while!!
+
+start_time <- Sys.time()
+
+# use amatch function to find title matches
+# Default method used for matching. Documentation for amatch function:
+# https://www.rdocumentation.org/packages/stringdist/versions/0.9.12/topics/amatch
+ title_matches <- cbind(
+  euctr_title_tm = EU_only_German$title,
+  euctr_id = EU_only_German$id,
+  IV_updated_titles[amatch(
+    EU_only_German$title_processed,
+    IV_updated_titles$title_processed,
+    maxDist = DISTANCE, matchNA = FALSE),]
+) |>
+  # keep only those where a match was found
+  filter(
+    !is.na(id)
+  ) |>
+  distinct(
+    # euctr_id, # we want to preserve all EUCTR crossreg for given trial
+    id,
+    .keep_all = TRUE
+  ) |>
```

Comment:
please look into implications of

- no distinct
- `distinct(id, .keep_all = TRUE)`
- `distinct(id, euctr_id, .keep_all = TRUE)`